incubator/tensorflow-serving/README.md

```diff
@@ -0,0 +1,159 @@
+tar -xzvf mnist-export.tar.gz
+rm -rf mnist-export.tar.gz
+cd /
+```
+
+* You will see that the contents of the model are stored in the  directory. This is the first version of the model that we will serve.
+
+```
+tree /serving/model/mnist
+/serving/model/mnist
+└── 1
+    ├── saved_model.pb
+    └── variables
+        ├── variables.data-00000-of-00001
+        └── variables.index
+
+umount /serving
+```
+
+## Create Persistent Volume
+
+Creating Persistent Volume with configuration like below
+
+```
+--- 
+apiVersion: v1
+kind: PersistentVolume
+metadata: 
+  labels: 
+    model: mnist
+  name: pv-nas-mnist
+spec:
+  persistentVolumeReclaimPolicy: Recycle
+  accessModes: 
+    - ReadWriteMany
+  capacity: 
+    storage: 5Gi
+  nfs:
+  	# FIXME: use the right IP
+    server: 10.244.1.4
+    path: "/serving/model/mnist"
+```
+
+## Prepare values.yaml
+
+* To deploy with GPU, you can create `values.yaml` like
+
+```
+---
+command: 
+  - /usr/bin/tensorflow_model_server
+args: 
+  - "--port=9090"
+  - "--model_name=mnist"
+  - "--model_base_path=/serving/model/mnist"
+image: "cheyang/tf-model-server-gpu:1.4"
+persistence: 
+  mountPath: /serving/model/mnist
+  pvc: 
+    matchLabels: 
+      model: mnist
+    storage: 5Gi
+```
+
+* To deploy without GPU, you can create `values.yaml` like 
+
+```
+---
+command: 
+  - /usr/bin/tensorflow_model_server
+args: 
+  - "--port=9090"
+  - "--model_name=mnist"
+  - "--model_base_path=/serving/model/mnist"
+image: "cheyang/tf-model-server:1.4"
+persistence: 
+  mountPath: /serving/model/mnist
+  pvc: 
+    matchLabels: 
+      model: mnist
+    storage: 5Gi
+```
+
+## Installing the Chart
+
+To install the chart with the release name `mnist`:
+
+```bash
+$ helm install --values values.yaml --name mnist incubator/acs-tensorflow-serving
+```
+
+## Uninstalling the Chart
+
+To uninstall/delete the `mnist` deployment:
+
+```bash
+$ helm delete mnist
+```
+
+The command removes all the Kubernetes components associated with the chart and
+deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the Service Tensorflow Serving
+chart and their default values.
+
+| Parameter | Description | Default |
+|-----------|-------------|---------|
+| `image` | TensorFlow Serving image | `registry.cn-hangzhou.aliyuncs.com/tensorflow-samples/tensorflow-serving:1.4.0-devel-gpu` |
+| `imagePullPolicy` | `imagePullPolicy` for the service mnist | `IfNotPresent` |
+| `port` | Tensorflow server port | `9090` |
+| `serviceType` | The service type which supports NodePort, LoadBalancer | `LoadBalancer` |
+|`replicas`| K8S deployment replicas | `1` |
+|`gpuCount`| The gpu resource to claim, if it's 0, means no gpu  | `1` |
+| `Cpu`    | container requested cpu            | `100m`                                                            |
+| `Memory` | container requested memory         | `512Mi`                                                           |
```

Comment:
variables should start lowercased, but with regards to resources it's fine to just add a `resources` variable instead that user can fill with their own request and limit settings. Might be easier.

Reply:
You also don't actually set them in the Deployment, only the gpu.